pkg/controllers/allocation/controller.go

```diff
@@ -105,5 +127,50 @@ func (c *Controller) Reconcile(ctx context.Context, object client.Object) (recon
 			zap.S().Errorf("Continuing after failing to bind, %s", err.Error())
 		}
 	}
-	return reconcile.Result{RequeueAfter: c.Interval()}, nil
+	return reconcile.Result{}, nil
+}
+
+// Watches returns the necessary information to create a watch
+//   a. source: the resource that is being watched
+//   b. eventHandler: which controller objects to be reconciled
+//   c. predicates: which events can be filtered out before processed
+func (c *Controller) Watches(ctx context.Context) (source.Source, handler.EventHandler, builder.WatchesOption) {
+	return &source.Kind{Type: &v1.Pod{}},
+		handler.EnqueueRequestsFromMapFunc(handler.MapFunc(
+			func(obj client.Object) []reconcile.Request {
+				pod := obj.(*v1.Pod)
+				if pod.Spec.NodeName != "" || podutils.IsOwnedByDaemonSet(pod) {
+					return nil
+				}
+				provisioner, err := c.filter.GetProvisionerFor(ctx, pod)
+				if err != nil {
+					zap.S().Errorf("Retrieving provisioner, %s", err.Error())
+					return nil
+				}
+				c.batcher.Add(fmt.Sprintf("%s/%s", provisioner.Name, provisioner.Namespace))
```

Comment:
Consider client.ObjectKeyForResource

Reply:
I'm not sure where this is? Did you mean `client.ObjectKeyFromObject`?  Either way, I changed the API to accept a client.Object and am pulling off name/namespace within the Batcher now.

Reply:
Is it possible to use a runtime.Object and avoid the controller-runtime dependency? It's an inner type of client.Object anyways.

Reply:
okay, I can use `k8s.io/apimachinery/pkg/apis/meta/v1/Object` I don't think I can use runtime.Object though..

Reply:
I always confused those two.